analysis/org.eclipse.tracecompass.analysis.profiling.ui/src/org/eclipse/tracecompass/analysis/profiling/ui/views/flamechart/FlameChartView.java

```diff
@@ -741,28 +749,38 @@ private void updateConfigureSymbolsAction() {
     }
 
     private void resetSymbols() {
-        List<TimeGraphEntry> traceEntries = getEntryList(getTrace());
-        if (traceEntries != null) {
-            for (TraceEntry traceEntry : Iterables.filter(traceEntries, TraceEntry.class)) {
-                ITimeGraphDataProvider<? extends TimeGraphEntryModel> provider = traceEntry.getProvider();
-                if (provider instanceof CallStackDataProvider) {
-                    ((CallStackDataProvider) provider).resetFunctionNames(new NullProgressMonitor());
+        try (FlowScopeLog fsl = new FlowScopeLogBuilder(LOGGER, Level.FINE, "Reseting Symbols").setCategory("Flame Chart").build();) { //$NON-NLS-1$ //$NON-NLS-2$
+            new Job("Reseting Symbols") { //$NON-NLS-1$
+                @Override
+                protected IStatus run(@Nullable IProgressMonitor monitor) {
+                    try (FlowScopeLog fsl2 = new FlowScopeLogBuilder(LOGGER, Level.FINE, "Reseting Symbols - inner").setParentScope(fsl).build()) { //$NON-NLS-1$
```

Comment:
Ideally, this part should be refactored into a separate function as we have 7 indentation levels.